### DISCUSSION TITLE
[WIP] Add `/sample` endpoint

### DISCUSSION THREAD

**Discussion by claymcleod** [CBIIT/ccdi-federation-api#4] [API Design Discussion]
## Overview

To facilitate the querying of sample data (both line level and summary), I propose the introduction of a `/sample` endpoint and subsequent namespace for the API. This endpoint requires special consideration because

- It would be the first API endpoint in the spec, setting some conventions that other API endpoints should follow, and
- I anticipate this would be one of the most heavily used API endpoints.

## Approach

There are many different ways I considered proposing this endpoint—especially since this will set the convention for other endpoints in the future. After some thought, I settled on the following:

- **`GET /sample`.** This is the primary endpoint for retrieving line-level data. In short, this endpoint would, by default, return all samples and available fields for all line-level data. Support will be provided for only returning particular fields of interest if desired. The data elements that are returned (and, subsequently, are available as filters) would be an ever growing list of metadata elements as defined by the harmonization subgroup. Note that, for sites that cannot provide line-level data, responses will be defined to indicate that no data is available (e.g., HTTP status code 204 or `No Content`).
- **`GET /sample/<id>`.** Returns the sample object for a particular id (id here refers to the `id` key in the sample object, which is a unique integer primary key that is specific to the source server's database).
- **`GET /sample/by/<element>`.** When aggregated counts are desired, users can use this endpoint to retrieve dynamically computed sample counts grouped by a particular data element. All organizations would be required to implement this endpoint. This should enable all available counting queries _besides_ just getting the total number of samples contained in the server. That functionality is covered by the first proposed endpoint.

### Considerations

- I considered including the summary information within the `/sample` API, but I think this will cause undue burden on many of the API endpoints—aggregating the summaries grouped by every data element available will be computationally burdensome to dynamically compute on each API query.
- There is an open question as to whether metadata values which can return more than one result should be plural or singular (`ethnicity` versus `ethnicities`). For the moment, I have chosen to propose keeping them singular, though I do not feel strongly about this.

### `/sample`

As described above, this endpoint represents the main endpoint for retrieving line-level data from the API. 

I propose the `/sample` endpoint return two top-level keys:

- `total`. Symbolizes the total number of unique samples contained within this API response, and
- `samples`. A list of sample objects (described below).

I propose each sample object contain the following keys to begin:

- `id`. A unique integer primary key that uniquely identifies the sample within the source server's database. This sequence should start at 1 and increment infinitely until the database runs out of space for each new sample. This will be incredibly useful in future API endpoints when you want to retrieve information on a particular samples (e.g., maybe we will have an endpoint like `/variants/sample/<id>`).
- `name`. The primary name/identifier used by the source server. This name should also be included in the list of identifiers under the `participant_id` key in the metadata object (discussed more later in this proposal).
- `metadata`. The ever-growing compendium of metadata values as defined by the data harmonization working group. I envision the definition of this object continuing to grow and be defined, but I've outlined four initial metadata entities that could be included in the first version later on in this proposal.

Valid HTTP response codes are:

- `200`. Line-level data was successfully returned within the API response.
- `204`. Line-level data is not serviceable by the API.
  - A blank JSON object, `{}`, is the only valid response body for this HTTP status code.

Other notes:

- Filters would be implemented using query parameters appended to the end of the URL (`?sex=F`).
- I envision heavily relying on HTTP status codes to indicate to the client whether information is available or not. For example, for source servers that cannot serve line level data, the HTTP status code 204 (`No Content`) should be issues along with an empty JSON body (`{}`).

The following represents a potential API response for the `/sample` endpoint.

```json
{
  "total": 214,
  "samples": [
    {
      "id": 1,
      "name": "SampleName",
      "metadata": {
        "sex": "F",
        "race": ["American Indian or Alaska Native"],
        "ethnicity": "Not Hispanic or Latino",
        "participant_id": ["SampleName", "Other ID"]
      }
    },
    ...
  ]
}
```

### `/sample/<id>`

Gets the sample object (defined in the last proposed API endpoint) for the specified ID. Here, "id" refers to the `id` key in the sample object, which is a unique integer primary key that is specific to the source server's database.

Valid HTTP response codes are:

- `200`. Object was found successfully and returned in the response.
- `404`. Object was not found within the database. 
  - Source servers which cannot show line level data should always return 404 for these requests, even if the client requests a valid `id`. The rationale is that the server should not acknowledge that _any_ ID exists to the client (for example, by returning a 200/204/403, etc).

For example, the request `/GET /sample/1` might return the following result:

```json
{
  "id": 1,
    "name": "SampleName",
    "metadata": {
      "sex": "F",
      "race": ["American Indian or Alaska Native"],
      "ethnicity": "Not Hispanic or Latino",
      "participant_id": ["SampleName", "Other ID"]
    }
}
```

### `/sample/by/<element>`

This endpoint represents the ability to group samples by a particular metadata element and count the number of samples with each corresponding value for that metadata element. 

I propose the top level object have the following keys:

- `total`. Symbolizes the total number of unique samples contained within this API response, and
- `values`. An object where keys represent the values for that metadata element contained within the source server and values represent integer counts of the number of samples with that value for the metadata element.

Valid HTTP response codes are:

- `200`. The request was well-formed, the metadata element was identified, the request was processed, and the results are returned in the API response.
- `422`. The metadata element for which the client requested to group by does not exist.

Other notes:

- The source server is only required to return metadata element values for which it has data corresponding to. For instance, if the source server does not contain any `F` values for sex, it is not required to include `"F": 0` in its response. A full list of acceptable values should be referred to the corresponding CDE.

For example, the request `GET /sample/by/sex` might return the following result:

```json
{
  "total": 158,
  "values": {
    "M": 76,
    "F": 82,
  }
}
```

### **`sex`** (6343385, v1.00, [Link](https://cadsr.cancer.gov/onedata/dmdirect/NIH/NCI/CO/CDEDD?filter=CDEDD.ITEM_ID=6343385%20and%20ver_nr=1))
 
This metadata element is defined by the caDSR as "Sex of the subject as determined by the investigator." In particular, this field does not dictate the time period: whether it represents sex at birth, sex at sample collection, or any other determined time point. Further, the descriptions for `F` and `M` suggest that this term can represent either biological sex, culture gender roles, or both. Thus, this field cannot be assumed to strictly represent biological sex. At present, the recommendation from the federation during our group discussion was to make this a singular field for each sample, as no one collected multiple values for sex—this is still open to feedback.

The following values are considered permissible values:

| Value | Long Name | Description |
|:-- | -- | -- |
| `U` | Unknown | Not known, not observed, not recorded, or refused. |
| `F` | Female | A person who belongs to the sex that normally produces ova. The term is used to indicate biological sex distinctions, or cultural gender role distinctions, or both. |
| `M` | Male | A person who belongs to the sex that normally produces sperm. The term is used to indicate biological sex distinctions, cultural gender role distinctions, or both. |
| `UNDIFFERENTIATED` | Intersex | A person (one of unisexual specimens) who is born with genitalia and/or secondary sexual characteristics of indeterminate sex, or which combine features of both sexes. |

### **`race`** (2192199, v1.00, [Link](https://cadsr.cancer.gov/onedata/dmdirect/NIH/NCI/CO/CDEDD?filter=CDEDD.ITEM_ID=2192199%20and%20ver_nr=1))
 
This metadata element is defined by the caDSR as "The text for reporting information about race based on the Office of Management and Budget (OMB) categories.". Upon examination of the large number of projects using the term, it appears to be the preferred term for the general concept of race. This field was recommended to be represented as an array of multiple values by the federation during our group discussion.

| Value | Long Name | Description |
|:-- | -- | -- |
| `Not allowed to collect` | Not Allowed To Collect | An indicator that specifies that a collection event was not permitted. |
| `Native Hawaiian or other Pacific Islander` | Native Hawaiian or Other Pacific Islander | Denotes a person having origins in any of the original peoples of Hawaii, Guam, Samoa, or other Pacific Islands. The term covers particularly people who identify themselves as part-Hawaiian, Native Hawaiian, Guamanian or Chamorro, Carolinian, Samoan, Chuukese (Trukese), Fijian, Kosraean, Melanesian, Micronesian, Northern Mariana Islander, Palauan, Papua New Guinean, Pohnpeian, Polynesian, Solomon Islander, Tahitian, Tokelauan, Tongan, Yapese, or Pacific Islander, not specified. |
| `Not Reported` | Not Reported | Not provided or available. |
| `Unknown` | Unknown | Not known, not observed, not recorded, or refused. |
| `American Indian or Alaska Native` | American Indian or Alaska Native  | A person having origins in any of the original peoples of North and South America (including Central America) and who maintains tribal affiliation or community attachment. (OMB) |
| `Asian` | Asian | A person having origins in any of the original peoples of the Far East, Southeast Asia, or the Indian subcontinent, including for example, Cambodia, China, India, Japan, Korea, Malaysia, Pakistan, the Philippine Islands, Thailand, and Vietnam. (OMB) |
| `Black or African American` | Black or African American  | A person having origins in any of the Black racial groups of Africa. Terms such as "Haitian" or "Negro" can be used in addition to "Black or African American". (OMB) |
| `White` | White | Denotes person with European, Middle Eastern, or North African ancestral origin who identifies, or is identified, as White. |

### **`ethnicity`** (2192217, v2.00, [Link](https://cadsr.cancer.gov/onedata/dmdirect/NIH/NCI/CO/CDEDD?filter=CDEDD.ITEM_ID=2192217%20and%20ver_nr=2.0))

This metadata element is defined by the caDSR as "The text for reporting information about ethnicity based on the Office of Management and Budget (OMB) categories." Upon examination of the large number of projects using the term, it appears to be the preferred term for the general concept of ethnicity.  At present, the recommendation from the federation during our group discussion was to make this a singular field for each sample, as no one collected multiple values for sex and the options appeared to be mutually exclusive—this is still open to feedback.

| Value | Long Name | Description |
|:-- | -- | -- |
| `Not allowed to collect` | Not Allowed To Collect | An indicator that specifies that a collection event was not permitted. |
| `Hispanic or Latino` | Hispanic or Latino | A person of Cuban, Mexican, Puerto Rican, South or Central American, or other Spanish culture or origin, regardless of race. The term, "Spanish origin," can be used in addition to "Hispanic or Latino." (OMB) |
| `Not Hispanic or Latino` | Not Hispanic or Latino | A person not of Cuban, Mexican, Puerto Rican, South or Central American, or other Spanish culture or origin, regardless of race. |
| `Unknown` | Unknown | Not known, not observed, not recorded, or refused. |
| `Not reported` | Not Reported | Not provided or available. |

### **`participant_id`** (6867052, v1.00, [Link](https://cadsr.cancer.gov/onedata/dmdirect/NIH/NCI/CO/CDEDD?filter=CDEDD.ITEM_ID=6867052%20and%20ver_nr=1))

This metadata element is currently under review, as the proposed CDE is listed as "retired" and is not recommended for future use.

**Comment by asafieva** [comment]
This is about endpoints naming.  
Can we consider Singular name convention as
/sample vs /samples
For many cases singular would be simpler. As an example
/diagnosis
Class names are singular in programming languages. This would make an easier mapping to classes in a programmatic API.

**Comment by e-t-k** [comment]
A few thoughts:

**`samples`**
- for  sites that cannot provide line-level data, how does someone retrieve the total number of samples available?

**`samples/by/<element>` endpoint**
- This endpoint is especially useful for sites that cannot provide line-level data--it will probably be the primary way that users of those sites explore their collections
- Can we include `participant_id` (or whatever we name it) and `name` as allowed elements for this endpoint? With non-line-level-able sites returning `404` for these queries. A lot of queries may be at the participant level and it would be useful to group samples by it.

**`id` keys**
- Having a sequential primary key means that different sites will use the same `id` to refer to different samples, and any frontend aggregating responses from multiple sites will need to namespace those `id`s. I don't think this is necessarily a problem.
- An alternative is to use UUIDs. (This doesn't mean that an implementation would need to use UUIDs as the primary key within the database but would just need one associated with each sample row.) However UUIDs are human-unfriendly and relatively long within the context of a URL.

**Singular vs plural**
- I'm in favor of using plural endpoint names when we are referring to a resource (ie sample or participant) , and singular when we are referring to a parameter (ie metadata element). So `samples/by/ethnicity`. ~~This is consistent with [REST best practices](https://apiguide.readthedocs.io/en/latest/build_and_publish/use_RESTful_urls.html).~~ Edit: Actually, best practices request that all nouns be plural, so this suggestion is _not_ consistent with that.

**Reply by claymcleod** [reply]
I've deleted my last comment because I made it while on-the-go and didn't really think through the issues. 

Here are my thoughts.

> for sites that cannot provide line-level data, how does someone retrieve the total number of samples available?

Good catch! In the original draft, I had included a `/samples/summary` endpoint that provided a lens into the most common summary metrics (in fact, this exact problem was the reason I first considered it). I ended up removing that endpoint from the draft, but I did not handle this case.

I would be okay reintroducing that endpoint, which appears to keep the cleanest functionality with respect to `/samples` returning a 204 in the case of non-line level data providers—I feel it would be confusing to respond with the "No Content" status code but then return the total field in the response body.

> Can we include participant_id (or whatever we name it) and name as allowed elements for this endpoint? With non-line-level-able sites returning 404 for these queries. A lot of queries may be at the participant level and it would be useful to group samples by it.

By default, I was thinking any metadata fields would be groupable here. In the case of `participant_id`, it would definitely be allowed. That does yield the question "what do we do when metadata fields are arrays: do we count every value in the array as one, or do we consider the array as a tuple?

I will mention, in the case of this endpoint specifically, grouping by name or participant I'd _hopefully_ only returns one count per value (or else, you have multiple samples with the same id/name. But, since a decision here likely sets the precedent for future endpoints where grouping by these elements will be more useful, I think it's prudent to allow it.

**id keys**

I agree with what you lay out here. I tend to lean towards using an integer id that is only unique to the source server—UUIDs are relatively unfriendly to work with as you say, and there is also the sneakily important motivation that we don't intend on making a statement for a "universal id" system through this API. If you use a UUID, there is always the chance the community will adopt that as the de-facto id for the sample throughout the community, whereas an integer id avoids that by its very nature of being non-unique throughout the community.

> I'm in favor of using plural endpoint names when we are referring to a resource (ie sample or participant) , and singular when we are referring to a parameter (ie metadata element). So samples/by/ethnicity. This is consistent with REST best practices. Edit: Actually, best practices request that all nouns be plural, so this suggestion is not consistent with that.

I think the point about all nouns being plural is in relation to the URL (at least, as I read it from the page you linked).

To the point about class names generally being singular, this is generally because classes, at a level of abstraction, do encapsulate a single entity (for instance, in Laravel, you may name your class `Sample`—that model will have properties for `name` and `id` (which apply to a single `Sample`), but the methods to access a collection are explicitly named as a multiplicity (e.g., `Sample::all()`).

Further, I think `/samples` is more appropriate for the line level data, because we are quite literally retuning a collection of samples. An argument could be made that accessing a single entity should be `/sample/1` (non-plural), but my feeling is this inconsistency is just asking to cause headaches downstream for consumers.

All in all, I'm in favor of plural nouns in the URLs, and I don't think best practices preclude us from using singular nouns when describing filters—especially in the case when the filter or attribute only has one value. The only question in my mind is when the attribute can have multiple values, such as `ethnicity` versus `ethnicities`. I slightly prefer consistency via singularity rather than being more accurate and plural, but my mind could be easily swayed on this.

**Comment by bfurner** [comment]
First of all, this is a really thorough and well-considered discussion, Clay, thanks. 

I suggest we may be getting ahead of ourselves a bit here as I don't know that we've come to any agreement on what we collectively understand the concept of "sample" to imply and we're building an API to return things we've not agreed upon the meaning of which seems problematic to me.  Is there a reason not to start with endpoints for `/participant` before moving downstream to other entities like sample?

**Reply by claymcleod** [reply]
Sure, that makes sense to me. My plan was to quickly knock out the conventions of the API using samples and then quickly turn around and make a very similar participants-centric endpoint after we finalized the details of how API endpoints should operate. However, I'm good with knocking out participants first since that is really what the group is focused on harmonizing right now: especially if you think there is more discussion to be had on exactly what a 'sample' means in this context.

Because the core pieces of the proposal (namely, how we organize endpoints) is somewhat entity agnostic, I can essentially change the name from `samples` to `subjects`/`participants` and that will be the end of the needed changes to accommodate this.

To that end, should these be called 'subjects' or 'participants'? I typically think of 'subjects' as a term that fully encapsulates but is broader than the term 'participants'—to future proof things, we have historically used the term 'subjects' to better represent the variety of sources (e.g., cell lines that have been purchased). However, I suppose one could make the argument that, even in the case of cell lines, those individuals are 'participating' in research in general, and that is really the scope of this API.

I don't have a strong opinion and am open to either.

**Reply by GeoffLyle** [reply]
Information on how Treehouse stores data:
We store information on a `donor` and `sample` level.  In this case I believe our `donor` would match up to `subject` or `participant`.   There can be multiple samples from the same `donor`, in some cases we may have PDX and cell lines derived from a patient who also has samples from primary and metastasis sites.  Our view of a `sample` is any RNA-Seq data file that we have collected and intended to run through our gene expression quantification pipelines. 

I don't feel strongly about `subject` vs `participant` as we do not use either of those terms.

**Comment by asafieva** [comment]
This is about a resource endpoint "participant". Here is a suggested resource definition. If this is acceptable, we can start a git branch to incorporate definitions into yaml. The expectation is that study_id and participant_id is a unique combination across the federation.
    participant:
        participant_id:
          type: string
          example: CCDI.16004
        study_id:
          type: string
          example: phs008844
        study_arm_id:
          type: string
          example: phs008844-arm1
        race:
          type: string
          example: American Indian or Alaska Native
        gender:
          type: string
        sex:
          type: string
        ethnicity:
          type: string
          example: Hispanic or Latino
        alternate_participant_id:
          type: string
      required:
        - type
        - study_id
        - participant_id
        - race
        - gender
        - ethnicity

**Reply by claymcleod** [reply]
Yeah I absolutely think this is along the right track. It would be interesting to hear your perspective on the "participant" versus "subject" question.

On another note, I feel quite strongly that we don't want to create _another_ universal identifier (e.g., `CCDI.16004`). When I think about the uniqueness of `study_id` and `participant_id` as you proposed, I think it's quite possible that those would _not_ be unique amongst the federation, because

- The participant ID is simply the _primary_ identifier that the source server has given the participant, and
- The study ID is very likely going to link to some common identifier, such as the dbGaP identifier that you referenced in your example.

Thus, I can see two sites downloading and processing data from the same dbGaP submission, just calling the sample whatever it was called in the dbGaP download, and thus creating a situation where this pair is non-unique.

**Reply by asafieva** [reply]
As of "participant" versus "subject" - I see "participant" is used in clinical trial contexts.
<p>About ID uniqueness. My question was: could a <b>combination</b> [study_id::participant_id] be unique across the federation? 
<p>Can any <b>combination of IDs</b> which includes study_id, as [study_id, sample_id] be <b>unique</b>? [study_id, other_resource_id]?
<p>This text I used as participant_id was completely arbitrary for yaml sake; we can put anything in yaml as an example, 'IID_H168421' if decided this to be a <I>string</i>.

**Comment by grugna** [comment]
Would it make sense to differentiate from No content and No access, so 403 in case the source has the data but cannot share it vs 204 or 404 for lack of the data?
I have to double check with @bfurner but I think we can share the IDs, as far as the cohort has more than 5 subjects. This could be a way to know what participant has data in another system and down the line with CPI/MPI we would already be setup to make a connection.
For data that are protected I am thinking of something like this as a response with 403
```
{
  "total": 214,
  "samples": [
    {
      "id": 1
    },
    {
      "id": 2
    },
    ...
  ]
}
```


I also think in the response beside total and metadata we should add a source information, like a URL that the user can reach to ask for access to the protected data.
```
{
  "total": 214,
  "source": https://www...."
  "samples": [
    {
      "id": 1
    },
    {
      "id": 2
    },
    ...
  ]
}
```

As far as the IDs, we are not using integers, We can for sure generate some integer value for this project, but I think it should be in a string format and allow each group to use their current IDs vs generating a new ones.

**Reply by claymcleod** [reply]
With respect to your question about 403 versus 204, it's an interesting philosophical question from an API perspective. My personal feeling is that we should representing _only_ what data this major version of the API is ever willing to provide—not necessarily reflecting what data exists in the host's database.

To put it another way,

- If the situation is that the API _can_ return line-level data, but only if you are an authorized individual, then I feel the API should return a 403.
- If the situation is that the API _will never_ return line-level data, even though the data is technically available within the backend endpoint, then I feel the API should return a 204.

Since lack of authorization is explicitly part of the requirements for this version of the API, I would argue that a 204 response is more appropriate—again, only for this major version of the API. If we introduce the capability to authorize a user in later versions, then I would say we should switch the response to a 403.

Sidenote: if no other reason that convention, it does feel strange to me to return data within a non-successful response code.

> As far as the IDs, we are not using integers, We can for sure generate some integer value for this project, but I think it should be in a string format and allow each group to use their current IDs vs generating a new ones.

Okay, that is interesting: I will have to consider that.

**Reply by melissacline** [reply]
I agree with @claymcleod .  If there are data that cannot be returned for security reasons (e.g. fewer than five participants in the cohort), that's fundamentally different from the situation when there are no data.  For example, the addition of one more participant could significantly change what are returned.  For a rare disease application (childhood cancers being rare diseases), it sounds more useful to return a 204 if there are data that cannot be shared - even if it might be more frustrating to the end user.

**Reply by grugna** [reply]
@claymcleod and @melissacline your point makes 100% sense to me. However, as Mark Cunningham often repeated their application wants to focus on pointing the user to where the data are. And if we return 204 without any information about the data being present at one of the end node we lose this opportunity and the user will never know there are data points in the platform that returned 204. 

How are you thinking of approaching this?

**Reply by claymcleod** [reply]
Actually, this is a really great question. Let's assume that the API returns links to where you can access the data (or information on where the data came from). In the case where you cannot return row-level data, what would you expect to return there? Just a single link that explains where the cohort came from? Or maybe you can actually share identifiers but you just can't return row-level _metadata_ information?

**Reply by melissacline** [reply]
Actually, I'm changing my mind.  If the API returns the following:

- Data when it can
- 204 when nothing can be returned because there are no matching records
- 403 when there are matching records, but they cannot be returned for regulatory reasons
then in the case of 403, at least the user would know that there are more data available, and that can potentially offer a path forward in finding the data.  

Especially for protected data (such that nothing can be returned other than a 403), we may need to think about _how much_ metadata we can share without effectively divulging the sample.  This was the trap that Beacon fell into in the Carlos Bustamante paper: if you construct a query (or a series of queries) that are so specific to one sample that they can (all) be matched only if the repository contains the sample, then there are instances where those query responses add up to a data leak.   If this sounds like a valid concern, then maybe we should start addressing it by looking at the data sharing policies in question.

**Reply by grugna** [reply]
I agree @melissacline as policy we can share the USI responding to a query but we block the responses that have cohort smaller than 5 subjects since hypothetically you could reconstruct part of the subject data reducing all the cohorts to 1. 
For this reason returning just the IDs would be inconsistent, so it is better in my opinion to return a 403 with information on where to potentially get access to the extra data.

@claymcleod this will for sure add some complexity on who will aggregate the queries since they will need to find a nice way to display all these information in a scenario where for instance one place returns data and one returns 403. But I think that is not in the scope of this project.

What do you think?

**Comment by melissacline** [comment]
Remembering the TCGA metadata, which has its whole hierarchy of Participant
-> Sample -> Vial -> Portion -> Analyte and so forth, the Treehouse data
would seem well-described with participant_id.  If we stick with
participant_id rather than sample_id, we can save sample_id for later when
we might want to distinguish between the PDX and the cell lines of the same
patient.

**Reply by asafieva** [reply]
Can a _sample_ (sample_id) resource in API always include information about _study_ and _participant_ (links to participant_id and study _id)?

**Reply by claymcleod** [reply]
To your point, I think the concept of "sample" as I proposed it here does encapsulate the differences between PDX and cell lines of the same individual. For the "participant" entity, I've created a [`/subject`](https://github.com/CBIIT/ccdi-federation-api/discussions/7) discussion now.

> Can a sample (sample_id) resource in API always include information about study and participant (links to participant_id and study _id)?

Yes, at least in the way I envision it, a sample is always a child of a subject, and a subject is likely linked to a study (though that hierarchy is still up for debate, just stating those links will be there in some way/shape/form).

**Comment by melissacline** [comment]
It sounds wise to always include information on the study and
participant/subject when returning a sample ID.  Or putting it in other
words, to assume that combinations of (study, subject, sample) are unique
but not assume uniqueness in (study, sample).

Given how "participant" is used in clinical trial contexts, I think that
Clay is wise to suggest that we use "subject", since it's not as heavily
overloaded and better supports "future-proofing" this API.  I'm not hearing
a compelling case for "participant" over "subject".

**Reply by claymcleod** [reply]
From what I heard, the case for participant is, in essence, associated with the fact that focus groups have found the terms "donor" and "subject" dehumanizing. I outlined the concerns in [this discussion](https://github.com/CBIIT/ccdi-federation-api/discussions/7) and struck what I felt was a solid compromise to maintain the human element while also remaining as accurate as possible. Take a look and let me know what you think!

**Comment by claymcleod** [comment]
Addressed in #16.
